util/Java.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2021 Grakn Labs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package grakn.common.util;
+
+public class Java {
+    public static Integer getMajorVersion() {
+        String version = System.getProperty("java.version");
+        if (version.startsWith("1.")) {
```

Comment:
The logic seems quite involved for something as simple as getting the version number. Are we sure it works across various implementations, eg., OpenJDK, Oracle JDK in all operating systems?

Is there a way to make it simpler?

Reply:
@lolski the reason for the complexity is that the format of the version has changed between Java <= 8 (`1.8`) and Java >= 9 (`9.0`). I'm pretty confident it works as I've tested it under both OpenJDK and Oracle Java of versions 8 and 11. 
Also, the property is _guaranteed_ to exist under whatever JVM you're running.

Reply:
@vmax will the code throw an unexpected exception if the format isn't what we expect? if so, let's update the code so that it throws a nice error: `Unrecognised Java version: ''.`

Reply:
@lolski addressed in https://github.com/graknlabs/common/pull/99/commits/8a29de5eb82089afb53b7cb28dd860d21bc54ad0